service/arm_templates/public_load_balancer_setup.json

```diff
@@ -0,0 +1,137 @@
+      }
+    },
+    {
+      "apiVersion": "[parameters('loadBalancersAPIVersion')]",
+      "name": "[variables('loadBalancerName')]",
+      "type": "Microsoft.Network/loadBalancers",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Network/publicIPAddresses/', variables('loadBalancerPublicIPName'))]"
+      ],
+      "properties": {
+        "frontendIPConfigurations": [
+          {
+            "name": "[variables('loadBalancerFrontendName')]",
+            "properties": {
+              "publicIPAddress": {
+                "id": "[variables('loadBalancerPublicIPId')]"
+              }
+            }
+          }
+        ],
+        "backendAddressPools": [
+          {
+            "name": "[variables('loadBalancerBackendPoolName')]"
+          }
+        ],
+        "copy": [
+          {
+            "name": "loadBalancingRules",
+            "count": "[length(parameters('ports'))]",
+            "input": {
+              "name": "[concat('loadBalancingRule-', string(add(copyIndex('loadBalancingRules'), 1)))]",
+              "properties": {
+                "frontendIPConfiguration": {
+                  "id": "[variables('loadBalancerFrontendID')]"
+                },
+                "backendAddressPool": {
+                  "id": "[variables('loadBalancerBackendPoolID')]"
+                },
+                "protocol": "Tcp",
+                "frontendPort": "[parameters('ports')[copyIndex('loadBalancingRules')].frontend]",
+                "backendPort": "[parameters('ports')[copyIndex('loadBalancingRules')].backend]",
+                "enableFloatingIP": false,
+                "probe": {
+                  "id": "[concat(variables('loadBalancerProbeID'), '-', string(add(copyIndex('loadBalancingRules'), 1)))]"
+                }
+              }
+            }
+          },
+          {
+            "name": "probes",
+            "count": "[length(parameters('ports'))]",
+            "input": {
+              "name": "[concat(variables('loadBalancerProbeName'), '-', string(add(copyIndex('probes'), 1)))]",
+              "properties": {
+                "protocol": "Tcp",
+                "port": "[parameters('ports')[copyIndex('probes')].frontend]",
+                "intervalInSeconds": 5,
+                "numberOfProbes": 2
+              }
+            }
+          }
+        ]
+      }
+    }
+  ],
+  "outputs": {
+    "loadBalancerBackendPoolID": {
+      "type": "object",
+      "value": {
+        "id": "[variables('loadBalancerBackendPoolID')]",
+        "name": "[variables('loadBalancerBackendPoolName')]"
+      }
+    },
+    "loadBalancerIDOutput": {
+      "type": "string",
+      "value": "[reference(variables('loadBalancerID'), parameters('loadBalancersAPIVersion'), 'Full')]"
+    },
+    "PublicIPIDOutput": {
+      "type": "string",
+      "value": "[reference(variables('loadBalancerPublicIPId'), parameters('publicIPAddressesAPIVersion'), 'Full')]"
+    }
```

Comment:
Do we need those two last outputs?

Reply:
They were for testing purposes, I'm done with the testing and removed them :)